test/Maestro.ScenarioTests/TestParameters.cs

```diff
@@ -19,21 +20,54 @@ public class TestParameters : IDisposable
     {
         internal readonly TemporaryDirectory _dir;
 
-        public static async Task<TestParameters> GetAsync()
+        private static string[] maestroBaseUris;
+        private static int maestroBaseUriIndex = 0;
+        private static string maestroToken;
+        private static string githubToken;
+        private static string darcPackageSource;
+        private static string azdoToken;
+        private static SemaphoreSlim mutex;
+
+        static TestParameters()
         {
             IConfiguration userSecrets = new ConfigurationBuilder()
                 .AddUserSecrets<TestParameters>()
                 .Build();
 
-            string maestroBaseUri = Environment.GetEnvironmentVariable("MAESTRO_BASEURI") ??  userSecrets["MAESTRO_BASEURI"] ?? "https://maestro-int.westus2.cloudapp.azure.com";
-            string maestroToken = Environment.GetEnvironmentVariable("MAESTRO_TOKEN") ?? userSecrets["MAESTRO_TOKEN"];
-            string githubToken = Environment.GetEnvironmentVariable("GITHUB_TOKEN") ?? userSecrets["GITHUB_TOKEN"];
-            string darcPackageSource = Environment.GetEnvironmentVariable("DARC_PACKAGE_SOURCE");
-            string azdoToken = Environment.GetEnvironmentVariable("AZDO_TOKEN") ?? userSecrets["AZDO_TOKEN"];
+            maestroBaseUris = (Environment.GetEnvironmentVariable("MAESTRO_BASEURIS")
+                    ?? userSecrets["MAESTRO_BASEURIS"]
+                    ?? "https://maestro-int.westus2.cloudapp.azure.com")
+                    .Split(',');
+            maestroToken = Environment.GetEnvironmentVariable("MAESTRO_TOKEN") ?? userSecrets["MAESTRO_TOKEN"];
+            githubToken = Environment.GetEnvironmentVariable("GITHUB_TOKEN") ?? userSecrets["GITHUB_TOKEN"];
+            darcPackageSource = Environment.GetEnvironmentVariable("DARC_PACKAGE_SOURCE");
+            azdoToken = Environment.GetEnvironmentVariable("AZDO_TOKEN") ?? userSecrets["AZDO_TOKEN"];
+            mutex = new SemaphoreSlim(1);
+        }
+
+        private static string GetMaestroBaseUri()
+        {
+            var maestroBaseUri = maestroBaseUris[maestroBaseUriIndex];
+            maestroBaseUriIndex =
+                maestroBaseUriIndex == maestroBaseUris.Length - 1
+                ? 0
+                : maestroBaseUriIndex + 1;
+            return maestroBaseUri; 
+        }
+
+        public static async Task<TestParameters> GetAsync()
+        {
+            
 
             var testDir = TemporaryDirectory.Get();
             var testDirSharedWrapper = Shareable.Create(testDir);
 
+            await mutex.WaitAsync();
+
+            var maestroBaseUri = GetMaestroBaseUri();
```

Comment:
It would be maybe better if some tests had an attribute which would say "use other URI"?

Reply:
I think we were saying that it wouldn't be random which tests use what so that it's easy to tell if a failure might be caused by the URL

Reply:
yup, we did, I added a parameter to the method that initializes the Test Parameters